### ISSUE TITLE
[tf.js][WebGPU] Pure GPU pipeline for running inference and presenting results on html5 canvas 

### ISSUE THREAD

**Issue by bernii** [tensorflow/tfjs#7701]
Hi!

This is related to https://github.com/tensorflow/tfjs/issues/6677 - I originally posted there but maybe separate issue if more appropriate. 

While working with tf.js and running a swin transformer inference with tf.js/webgpu I noticed that `toPixels` method of drawing to canvas is really slow (especially for high-res images).

I tried to use the pure GPU method that is presented in the [gpu pipeline example](https://github.com/tensorflow/tfjs-examples/tree/master/gpu-pipeline) but not having much luck. I tried to incorporate the changes that @rahul-lokesh mentioned in https://github.com/tensorflow/tfjs/issues/6677  but that didn't solve the problem either.

I uploaded the code in case you want to take a look https://img-cut.aishoot.co/puregpu.html (you need to reload the page before starting the pipeline again, wanted to keep code as simple as possible)

This is what I get when using standard `toPixels` and placing things on canvas after getting the inference output from the model (RGB image).
![Screenshot 2023-05-18 at 10 51 54](https://github.com/tensorflow/tfjs/assets/778717/011d6cc9-60be-474d-9c7f-6599a721618e)
and this is what I get when using the `pure GPU` approach
![Screenshot 2023-05-18 at 10 52 16](https://github.com/tensorflow/tfjs/assets/778717/90e7ed12-ff8a-4203-a047-94e90ab08886)

Seems that creating custom backend creates some problems with inference itself.

I see two major problems
- there are tons of errors in the console (also from model inference - zero errors before), plus the result from inference is a bit different vs using standard 'webgpu' backend
- image is kinda repeated - is this a shader issue?

I got kinda stuck at this point but happy to provide more info or do more debugging 🤔

**Comment by qjia7** [comment]
@bernii The error is because the buffer size exceeds the max buffer limits. Please refer the code [here](https://github.com/tensorflow/tfjs/blob/master/tfjs-backend-webgpu/src/base.ts#L53) when you create a custom webgpu backend, which will allow  you to use the hardware supported max buffer size.

**Comment by qjia7** [comment]
Another way is to use the default webgpu backend instead of the custom one. You can get the tfjs-webgpu's `device` by `device = (tf.backend() as WebGPUBackend).device;`, and use this `device` to do post-processing.

**Comment by bernii** [comment]
@qjia7 thanks for help!  using the default backend and getting `tfjs-webgpu`'s `device` fixed all the memory issues!

I'll keep on investigating the image 'clones' being rendered on the canvas problem - possibly a bug in the shader code
